### ISSUE TITLE
Mermaid support for rendering SVG

### ISSUE THREAD

**Issue by bsmth** [mdn/yari#10497]
I'd like to create a tracking issue for Mermaid support as the best candidate for maintaining SVG diagrams. There are a number of discussions about this:

__Discussions:__

- ["Exempt SVGs from minifying"](https://github.com/mdn/yari/discussions/3652)
- ["Using SVG format instead of `.png` images"](https://github.com/orgs/mdn/discussions/631)
- ["Support Mermaid in MDN Markdown"](https://github.com/orgs/mdn/discussions/358)

And as for some options, if we don't want to add more JS client side or do the rendering at build time as a Yari dependency, there is the option to generate SVG from mmd files in an automated way [using the CLI](https://github.com/mdn/content/pull/31812#issuecomment-1929337512) (a CI job / script / automation in content repo on new `.mmd` files, for instance).

e.g.;

```bash
npm install -g @mermaid-js/mermaid-cli
mmdc -i input.mmd -o output.svg
```

In terms of optimizations, there is [SVGO](https://github.com/svg/svgo) which also [has a GUI](https://github.com/svg/svgo).

__Related issues:__

- https://github.com/mdn/yari/issues/1678

__Pull requests:__

- https://github.com/mdn/content/issues?q=label%3A%22reduce+binary+assets%22+
- https://github.com/mdn/content/pull/4505

**Comment by caugner** [comment]
@bsmth How about if the shared-assets repo would support Mermaid and automatically render/deploy them?

**Comment by bsmth** [comment]
> [@bsmth](https://github.com/bsmth) How about if the shared-assets repo would support Mermaid and automatically render/deploy them?

Yeah I think this is another good option where we have build / optimization scripts in that repo that are completely decoupled from content / rari etc.
